docs/data/toolpad/core/components/sign-in-page/sign-in-page.md

```diff
@@ -0,0 +1,15 @@
+---
+productId: toolpad-core
+title: Sign In Page
+components: SignInPage, Account
+---
+
+# Sign In Page
+
+<p class="description">A component that renders a sign in page for your application.</p>
+
+{{"demo": "BasicSignInPage.js"}}
```

Comment:
This page needs more context. An explanation what is being demoed on the page, why you'd use it, how it interacts with the app provider, how you'd use it to solve different problems...

I propose a general structure:
* In the general demo at the top, that shows off the sign in page, I'd show only 3 oath providers. I'd leave out the credentials provider in this one.
* It needs a section that explains setting it up with an oauth provider. The example here shows only one provider. It also needs to explain which other oauth providers are available to be used with this component.
* It needs a section that explains setting it up with the credentials provider. The example shows the credentials provider only.
* It needs a section on customization. And how it interacts with branding in the App provider. If deep customization is unfinished, add an explanation on how it will work and mark it as being under construction.
* We keep the previous sections agnostic of the auth library, but we add a section on how you'd integrate with next-auth (doesn't necessarily need an embedded demo, but we could link to an example)
* We add a section on how to integrate with next.js pages router and next.js app router (doesn't necessarily need an embedded demo, but we could link to an example)